Source/engine/demomode.cpp

```diff
@@ -57,53 +57,50 @@ void PumpDemoMessage(DemoMsgType demoMsgType, uint32_t message, int32_t wParam,
 	Demo_Message_Queue.push_back(msg);
 }
 
+template <class T>
+T ReadFromStream(std::ifstream &stream)
+{
+	T value;
+	stream.read(reinterpret_cast<char *>(&value), sizeof(value));
+	return value;
+}
+
+template <class T>
+void WriteToDemo(T value)
+{
+	DemoRecording.write(reinterpret_cast<char *>(&value), sizeof(value));

```

Comment:
This is an unaligned write and endian-dependent
Please do writes and reads the same way we do them for other things, e.g saves